app/scripts/models/mixins/resume-token.js

```diff
@@ -44,16 +46,30 @@ define(function (require, exports, module) {
     },
 
     /**
-     * Sets model properties from a ResumeToken.
+     * Sets model properties from a ResumeToken, unless said token
+     * is invalid according to `this.resumeTokenSchema`.
      *
      * @method populateFromResumeToken
      * @param {ResumeToken} resumeToken
      */
     populateFromResumeToken: function (resumeToken) {
       if (this.resumeTokenFields) {
-        this.set(resumeToken.pick(this.resumeTokenFields));
+        var pickedResumeToken = resumeToken.pick(this.resumeTokenFields);
+
+        if (isResumeTokenValid(pickedResumeToken, this.resumeTokenSchema)) {
+          this.set(pickedResumeToken);
+        }
       }
     }
   };
+
+  function isResumeTokenValid (resumeToken, schema) {
+    if (schema) {
+      var result = vat.validate(resumeToken, schema);
```

Comment:
YYYAAAASS!!!! Ref #3591. Does validation make sense within the ResumeToken model?